charts/prometheus-blackbox-exporter/templates/configmap.yaml

```
@@ -6,6 +6,10 @@ metadata:
   namespace: {{ template "prometheus-blackbox-exporter.namespace" . }}
   labels:
     {{- include "prometheus-blackbox-exporter.labels" . | nindent 4 }}
+  {{- if and .Values.secretConfig (not (empty .Values.secretAnnotations)) }}
```

why do you use "not empty"?
Imho it would work like this too?

```suggestion
  {{- if and .Values.secretConfig .Values.secretAnnotations }}
```